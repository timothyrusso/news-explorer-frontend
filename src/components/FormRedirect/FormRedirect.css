.form-redirect {
  text-align: center;
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  margin: 16px 0 28px;
}

.form-redirect_type_link {
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  border: none;
  background-color: #fff;
  color: #2F71E5;
  padding: 0;
}

.form-redirect_type_link:hover {
  cursor: pointer;
}
