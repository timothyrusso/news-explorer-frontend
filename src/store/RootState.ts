import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
