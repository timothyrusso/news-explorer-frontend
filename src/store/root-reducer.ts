import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
