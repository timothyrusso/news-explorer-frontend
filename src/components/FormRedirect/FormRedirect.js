import "./FormRedirect.css";

const FormRedirect = ({ onSwitch, popupRedirectText }) => {
  return (
    <p className="form-redirect">
      or{" "}
      <button
        aria-label="test"
        type="button"
        className="form-redirect_type_link"
        onClick={onSwitch}
      >
        {popupRedirectText}
      </button>
    </p>
  );
};

export default FormRedirect;
