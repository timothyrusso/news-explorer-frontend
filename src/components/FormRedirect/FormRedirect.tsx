import React, { FC } from 'react';
import { usePopup } from '../../hooks/usePopup';
import './FormRedirect.css';

type FormRedirectProps = {
  popupRedirectText: string;
};

const FormRedirect: FC<FormRedirectProps> = ({ popupRedirectText }) => {
  const { handleSwitchPopup } = usePopup();

  return (
    <p className="form-redirect">
      or{' '}
      <button
        aria-label="test"
        type="button"
        className="form-redirect_type_link"
        onClick={handleSwitchPopup}
      >
        {popupRedirectText}
      </button>
    </p>
  );
};

export default FormRedirect;
