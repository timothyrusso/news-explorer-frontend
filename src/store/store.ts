import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
