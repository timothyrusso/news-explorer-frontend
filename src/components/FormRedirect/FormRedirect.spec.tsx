import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import FormRedirect from './FormRedirect';

describe('FormRedirect component', () => {
  it('Should render the FormRedirect component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FormRedirect popupRedirectText={'Example text'} />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
