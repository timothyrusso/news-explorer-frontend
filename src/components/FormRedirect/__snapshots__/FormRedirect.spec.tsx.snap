// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRedirect component Should render the FormRedirect component 1`] = `
<DocumentFragment>
  <p
    class="form-redirect"
  >
    or 
    <button
      aria-label="test"
      class="form-redirect_type_link"
      type="button"
    >
      Example text
    </button>
  </p>
</DocumentFragment>
`;
